pkg/fanal/cache/key.go

```diff
@@ -50,7 +50,21 @@ func CalcKey(id string, analyzerVersions analyzer.Versions, hookVersions map[str
 		}
 	}
 
-	// TODO: add secret scanner option here
+	// Write secret config
```

Comment:
I think we can use the existing logic for misconfiguration.
https://github.com/aquasecurity/trivy/blob/6534dec6704fb010f84b6add5db355a5ee81b0a8/pkg/fanal/cache/key.go#L39C1-L51C3

```
	// Write policy, data contents and secret config file
	paths := append(artifactOpt.MisconfScannerOption.PolicyPaths, artifactOpt.MisconfScannerOption.DataPaths...)

	// Check if the secret config exists.
	if _, err := os.Stat(artifactOpt.SecretScannerOption.ConfigPath); err == nil {
		paths = append(paths, artifactOpt.SecretScannerOption.ConfigPath)
	}
	for _, p := range paths {
		hash, err := hashContents(p)
		if err != nil {
			return "", err
		}

		if _, err := h.Write([]byte(hash)); err != nil {
			return "", xerrors.Errorf("sha256 write error: %w", err)
		}
	}
```